stock_picking_type_shipping_policy/tests/test_stock_picking_type_shipping_policy.py

```diff
@@ -28,35 +29,46 @@ def setUpClass(cls):
             }
         )
 
+    def _run_procurement(self, product, qty):
+        moves_before = self.env["stock.move"].search([])
+        proc_group = self.env["procurement.group"]
+        uom = product.uom_id
+        proc_qty, proc_uom = uom._adjust_uom_quantities(qty, uom)
+        today = fields.Date.today()
+        proc_group = self.env["procurement.group"].create({})
+        values = {
+            "group_id": proc_group,
+            "date_planned": today,
+            "date_deadline": today,
+            "warehouse_id": self.warehouse or False,
+            "company_id": self.company,
+        }
+        procurement = proc_group.Procurement(
+            product,
+            proc_qty,
+            proc_uom,
+            self.customers_location,
+            product.name,
+            "PROC TEST",
+            self.company,
+            values,
+        )
+        proc_group.run([procurement])
+        moves_after = self.env["stock.move"].search([])
+        return moves_after - moves_before
+
+    def _validate_picking(self, picking):
+        picking.move_line_ids.write({"picked": True})
+        picking._action_done()
+
     def test_shipping_policy(self):
         self.pack_type.shipping_policy = "force_all_products_ready"
         self.pick_type.shipping_policy = "force_as_soon_as_possible"
-        # Create picking
```

Comment:
Rewrote this part to align with new default Odoo delivery flow that creates the Pick first, then Pack + Ship.
To ensure things are created as expected, we generate the initial move through a procurement.